flow-server/src/main/java/com/vaadin/flow/component/internal/UIInternals.java

```diff
@@ -594,7 +596,7 @@ private void registerDetachListenerForPendingInvocation(
         JavaScriptInvocationDetachListener listener = pendingJsInvocationDetachListeners
                 .computeIfAbsent(ownerNode, node -> {
                     JavaScriptInvocationDetachListener detachListener = new JavaScriptInvocationDetachListener();
-                    detachListener.registration = Registration.combine(
+                    detachListener.registration = combine(
                             () -> pendingJsInvocationDetachListeners
                                     .remove(node),
                             node.addDetachListener(detachListener));
```

Comment:
I would remove the `combine` method, as it introduces a lot of overhead for a single call (e.g, creating `List`s, as it expects potentially more than one registration)
Instead, I would simply create a lambda here

```suggestion
                    Registration registration = node.addDetachListener(detachListener);
                    detachListener.registration = () -> {
                        pendingJsInvocationDetachListeners.remove(node);
                        registration.remove();
                    };
```